### PR TITLE
#82729 - Propagate kv loading errors

### PR DESCRIPTION
**Pull request by tabman83**
Return proper error message when we fail to get password (or client secret) from KV - right now we get some generic error and it is unclear why it failed

### CODE REVIEW

**Review by dawidkeshop — APPROVED** [review]

File: src/CaptainHook.Common/CaptainHook.Common.csproj [hunk 1]
```diff
@@ -15,7 +15,7 @@
     <PackageReference Include="Autofac.ServiceFabric" Version="3.0.0" />
     <PackageReference Include="Azure.Identity" Version="1.2.2" />
     <PackageReference Include="Azure.Security.KeyVault.Secrets" Version="4.1.0" />
-    <PackageReference Include="Eshopworld.DevOps" Version="5.1.6" />
+    <PackageReference Include="Eshopworld.DevOps" Version="5.0.1" />
```
Comment (chomick) [inline src/CaptainHook.Common/CaptainHook.Common.csproj:18]: Do we know what was the actual issue there? Should we investigate bit more?

File: src/CaptainHook.DirectorService/DirectorService.cs [hunk 2]
```diff
@@ -88,7 +94,15 @@ protected override async Task RunAsync(CancellationToken cancellationToken)
                     _refreshInProgress = true;
                 }
 
-                _subscriberConfigurations = await LoadConfigurationAsync();
+                var loadConfigurationResult = await LoadConfigurationAsync();
+                if(loadConfigurationResult.IsError)
+                {
+                    // should log through bb;
```
Comment (chomick) [inline src/CaptainHook.DirectorService/DirectorService.cs:100]: Can you add the logging for that, please?

File: src/CaptainHook.Common/CaptainHook.Common.csproj (hunk 1 above)
Comment (tabman83) [reply src/CaptainHook.Common/CaptainHook.Common.csproj]: I have created a separate tech debt task for that and I would like to leave it out of the scope of this PR

**Review by chomick — APPROVED** [review]

File: src/CaptainHook.DirectorService/DirectorService.cs (hunk 2 above)
Comment (tabman83) [reply src/CaptainHook.DirectorService/DirectorService.cs]: Done

File: src/CaptainHook.Common/CaptainHook.Common.csproj (hunk 1 above)
Comment (tabman83) [reply src/CaptainHook.Common/CaptainHook.Common.csproj]: Also, downgrade is only shown here because of a merge from master

**Review by chomick — APPROVED** [review]

**Pull request merged**